src/compiler/moduleSpecifiers.ts

```diff
@@ -677,10 +749,10 @@ namespace ts.moduleSpecifiers {
 
         // Simplify the full file path to something that can be resolved by Node.
 
+        const preferences = getPreferences(host, userPreferences, options, importingSourceFile);

```

Comment:
Is this used outside the if block, or is it so cheap that it just doesn't matter where it is?

Reply:
I assume I had a reason to move it; maybe refactored away in later edits, but it is trivially cheap.